core/server/data/migrations/versions/4.35/2022-02-02-13-10-transform-specific-tiers-default-content-visibility.js

```diff
@@ -0,0 +1,114 @@
+const logging = require('@tryghost/logging');
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Checking default_content_visibility for specific tiers');
+
+        const settings = await knex('settings')
+            .select()
+            .whereIn('key', ['default_content_visibility', 'default_content_visibility_tiers']);
+        const contentVisibilitySetting = settings.find(d => d.key === 'default_content_visibility');
+        const visibilityTiersSetting = settings.find(d => d.key === 'default_content_visibility_tiers');
+        if (!contentVisibilitySetting || !contentVisibilitySetting.value) {
+            logging.warn('No default_content_visibility setting found.');
+            return;
+        }
+
+        if (!visibilityTiersSetting) {
+            logging.warn('No default_content_visibility_tiers setting found.');
+            return;
+        }
+
+        const contentVisibility = contentVisibilitySetting.value;
+        // Transform visibility to tiers when stored as nql string
+        if (!['public', 'members', 'paid'].includes(contentVisibility)) {
+            const isValidProductNqlFilter = /^(?:product:[\w-]+,?)+$/.test(contentVisibility);
+            let newVisibilityValue = 'paid';
```

Comment:
So am I right in saying that if `contentVisibility` is not one of `[public,members,paid]`, and it's not a valid product NQL filter, then we set `contentVisibility` to `paid` anyway?

Reply:
Or should the setting updates be inside the if-statement?

Reply:
> if contentVisibility is not one of [public,members,paid], and it's not a valid product NQL filter, then we set contentVisibility to paid anyway

@daniellockyer Yea that is correct, the reasoning behind this is to set the highest visibility restriction as default if someone in an unlikely scenario doesn't has the right NQL filter in place. We could as well set it to `public` instead as it only affects the default visibility, but my idea of using `paid` was to fallback to tighter visibility in case of an invalid value, wdyt ?

Reply:
Gotcha - makes sense 👍🏻